### DISCUSSION TITLE
Accessing dynamic variable name in github actions

### DISCUSSION THREAD

**Discussion by git-hub-007** [community/community#25634] [Actions]
<p>I want to access value of dynamic variable name which in effect is c_ip1, c_ip2 How do I construct this in github actions workflow ?<br>
Currently I am getting error if I access as below:</p>
<p>val=“needs.build.outputs.c_ip$ip”<br>
echo ${!val}</p>
<pre><code class="lang-auto">on: [push]

jobs:
  build:
    runs-on: ubuntu-latest
    outputs:
      c_ip1: ${{ steps.set_c.outputs.c_ip1 }}
      c_ip2: ${{ steps.set_c.outputs.c_ip2 }}

    strategy:
      matrix:
       include:
         - index: 1
           deploy: "dev" 
         - index: 2
           deploy: "prod" 
    steps:

      - id: set_c
        run: |
          res=10
          if [ ${{ matrix.index }} = "dev" ]; then
            echo "::set-output name=c_ip1::$res"
          else
            res=5
            echo "::set-output name=c_ip2::$res"
              
          fi
        continue-on-error: true     

  upgrade-c2:
    name: another job
    needs: build
    runs-on: ubuntu-latest

    strategy:
      matrix:
       include:
         - index: 1
           deploy: "dev" 
         - index: 2
           deploy: "prod" 

    steps:
    - run: |
         ip=${{ matrix.index }}
         val="needs.build.outputs.c_ip$ip"
         if [ ${{ matrix.deploy }} = "dev" ]; then
           #echo ${{ needs.build.outputs.c_ip[$ip] }}
           echo ${!val}
         else
           #echo ${{ needs.build.outputs.c_ip[$ip] }}
           echo ${!val}
         fi
      continue-on-error: 
</code></pre>

**Comment by Simran-B** [comment, marked as answer]
<p>GHA contexts are only accessible in workflow expressions <code>${{ }}</code>. Mixing them with shell commands like that will not work. I think you should do this:</p>
<p><code>echo ${{ needs.build.outputs[format('c_ip{0}', matrix.index)] }}</code></p>

**Reply by RichardTMiles** [reply]
I'd love to see this documented better; I feel the community doesn't have a grasp on this given my stack overflow searches.. (linked to this post)

**Reply by Darjow** [reply]
Thanks @Simran-B for this clean solution. 

For people wondering, this is also possible with secrets and vars. 
```${{ vars[format('{0}_CONTAINER_REGISTRY_DNS', env.PREFIX )] }}```

**Reply by PIrojahPerbak** [reply]
Cannot seem to get this to work with env vars?

**Comment by git-hub-007** [comment]
<p>Thanks it worked. Able to fetch the values.</p>

**Comment by RichardTMiles** [comment]
https://github.com/actions/runner/pull/2477

```yaml
   strategy:
      matrix:
        os: ['linux', 'windows','macos']
        arch: ['x86','x64']
    outputs:
      ${{ matrix.os }}_${{ matrix.arch }}_result: ${{ steps.step1.outputs.test }}
```
